### PR TITLE
Make "remove recent messages" more robust

### PR DESCRIPTION
**Pull request by bwindels**
Related: https://github.com/vector-im/riot-web/issues/10972
Requires: https://github.com/matrix-org/matrix-js-sdk/pull/1048

This just gets all the timelines known, and looks for events from the user in each, as opposed to going back through the links between individual timelines. The linked bug seemed to indicate that those links sometimes might get corrupted, and while this won't fix the timeline issues that result from that, it might make "remove recent messages" a bit more robust.

### CODE REVIEW

File: src/components/views/rooms/MemberInfo.js [hunk 1]
```diff
@@ -396,10 +395,10 @@ module.exports = createReactClass({
                     title: _t("Remove recent messages by %(user)s", {user}),
                     description:
                         <div>
-                            <p>{ _t("You are about to remove %(count)s messages by %(user)s. This cannot be undone. Do you wish to continue?", {count, user}) }</p>
+                            <p>{ _t("You are about to remove %(amount)s message(s) by %(user)s. This cannot be undone. Do you wish to continue?", {amount: amount, user}) }</p>
```
Comment (t3chguy) [inline src/components/views/rooms/MemberInfo.js]: for consistency `{amount, user}`

File: src/i18n/strings/en_EN.json [hunk 2]
```diff
@@ -804,9 +804,9 @@
     "No recent messages by %(user)s found": "No recent messages by %(user)s found",
     "Try scrolling up in the timeline to see if there are any earlier ones.": "Try scrolling up in the timeline to see if there are any earlier ones.",
     "Remove recent messages by %(user)s": "Remove recent messages by %(user)s",
-    "You are about to remove %(count)s messages by %(user)s. This cannot be undone. Do you wish to continue?|other": "You are about to remove %(count)s messages by %(user)s. This cannot be undone. Do you wish to continue?",
+    "You are about to remove %(amount)s message(s) by %(user)s. This cannot be undone. Do you wish to continue?": "You are about to remove %(amount)s message(s) by %(user)s. This cannot be undone. Do you wish to continue?",
     "For a large amount of messages, this might take some time. Please don't refresh your client in the meantime.": "For a large amount of messages, this might take some time. Please don't refresh your client in the meantime.",
-    "Remove %(count)s messages|other": "Remove %(count)s messages",
```
Comment (t3chguy) [inline src/i18n/strings/en_EN.json]: why are we moving away from proper pluralisation?

Comment (bwindels) [reply src/i18n/strings/en_EN.json]: because I couldn't be bothered figure out how it works I suppose. Is there a good example somewhere?

Comment (t3chguy) [reply src/i18n/strings/en_EN.json]: like it was I believe, the word `count` is special, it dictates whether to use `one` or `two` or `three` or `other` etc

Comment (bwindels) [reply src/i18n/strings/en_EN.json]: I see. I was having problems with the translation when using `count`, so was using it without knowing. The translation would be empty with `count=1`. How would I fix this, by adding a `...|one` string in `en_en.json`?

Comment (t3chguy) [reply src/i18n/strings/en_EN.json]: Yup, that should be it

**Comment by bwindels** [comment]
Branch needs to be rebased to make e2e tests pass? Looks like cider-by-default wasn't applied yet by the last run of the e2e tests.

**Pull request merged**